### PR TITLE
Add vector support to gRPC API

### LINKED ISSUE #2654: Add vector support to gRPC API
Since we added support for the experimental `vector` CQL data type in #2597, we can take advantage of this to add vector support to the gRPC API. This will also require documentation updates.

### PR DESCRIPTION
**Pull request by Hazel-Datastax**
**What this PR does**:
Added vector type codec to stargate grpc.
Add unit test for vector support in grpc。

**Which issue(s) this PR fixes**:
Fixes #2654 #2704

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by maheshrajamani — CHANGES_REQUESTED** [review]
Since this is for grpc api exposed to users, can you add some integration tests?
Current grpc integration tests can be found in this folder[here](https://github.com/stargate/stargate/tree/main/coordinator/testing/src/main/java/io/stargate/it/grpc)?

**Comment by Hazel-Datastax** [comment]
Yes, I am still working on it. Thank you for mentioning it.

**Comment by maheshrajamani** [comment]
> Yes, I am still working on it. Thank you for mentioning it.

Ah I didn't see the PR in draft.

File: coordinator/testing/src/main/java/io/stargate/it/grpc/VectorQueryTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    session.execute(
+        "CREATE TABLE IF NOT EXISTS vector_table (id int PRIMARY KEY, embedding vector<float, 5>)");
+    session.execute(
+        "CREATE CUSTOM INDEX embedding_index ON vector_table(embedding) USING 'StorageAttachedIndex'");
+  }
+
+  @Test
+  @DisplayName("Should be able to insert and fetch a vector value")
+  @Order(1)
+  public void insertAndGet(@TestKeyspace CqlIdentifier keyspace) {
```
Comment (maheshrajamani) [inline coordinator/testing/src/main/java/io/stargate/it/grpc/VectorQueryTest.java:55]: Can you also add some test for insert and select cases with binding variables? Similar to examples [here](https://github.com/stargate/stargate/blob/bab9b5c19dc586959f95fa170392082ef8e3e833/coordinator/testing/src/main/java/io/stargate/it/grpc/ExecuteQueryTest.java#L72C5-L79C38)

**Review by maheshrajamani — CHANGES_REQUESTED** [review]

**Comment by Yuqi-Du** [comment]
Need to change the size unmatched message.
[related](https://github.com/stargate/stargate/pull/2726/files)

**Review by maheshrajamani — APPROVED** [review]
LGTM.

**Pull request merged**